broadbean/sequence.py

```diff
@@ -0,0 +1,1065 @@
+            SpecificationInconsistencyError: If both f_cut and tau are given.
+        """
+
+        if kind not in ['HP', 'LP']:
+            raise ValueError('Filter kind must either be "LP" (low pass) or '
+                             '"HP" (high pass).')
+        if not isinstance(order, int):
+            raise ValueError('Filter order must be an integer.')
+        if (f_cut is not None) and (tau is not None):
+            raise SpecificationInconsistencyError('Can not specify BOTH a time'
+                                                  ' constant and a cut-off '
+                                                  'frequency.')
+
+        keystr = 'channel{}_filtercompensation'.format(channel)
+        self._awgspecs[keystr] = {'kind': kind, 'order': order, 'f_cut': f_cut,
+                                  'tau': tau}
+
+    def addElement(self, position: int, element: Element) -> None:
+        """
+        Add an element to the sequence. Overwrites previous values.
+
+        Args:
+            position (int): The sequence position of the element (lowest: 1)
+            element (Element): An element instance
+
+        Raises:
+            ValueError: If the element has inconsistent durations
+        """
+
+        # Validation
+        element.validateDurations()
+
+        # make a new copy of the element
+        newelement = element.copy()
+
+        # Data mutation
+        self._data.update({position: newelement})
+
+        # insert default sequencing settings
+        self._sequencing[position] = {'twait': 0, 'nrep': 1,
+                                      'jump_input': 0, 'jump_target': 0,
+                                      'goto': 0}
+
+    def addSubSequence(self, position: int, subsequence: 'Sequence') -> None:
+        """
+        Add a subsequence to the sequence. Overwrites anything previously
+        assigned to this position. The subsequence can not contain any
+        subsequences itself.
+
+        Args:
+            position: The sequence position (starting from 1)
+            subsequence: The subsequence to add
+        """
+        if not isinstance(subsequence, Sequence):
+            raise ValueError('Subsequence must be a sequence object. '
+                             'Received object of type '
+                             '{}.'.format(type(subsequence)))
+
+        for elem in subsequence._data.values():
+            if isinstance(elem, Sequence):
+                raise ValueError('Subsequences can not contain subsequences.')
+
+        if subsequence.SR != self.SR:
+            raise ValueError('Subsequence SR does not match (main) sequence SR'
+                             '. ({} and {}).'.format(subsequence.SR, self.SR))
+
+        self._data[position] = subsequence.copy()
+
+        self._sequencing[position] = {'twait': 0, 'nrep': 1,
+                                      'jump_input': 0, 'jump_target': 0,
+                                      'goto': 0}
+
```

Comment:
As stated previously, add optional arguments which are then used to create this dictionary